xwiki-platform-core/xwiki-platform-test/xwiki-platform-test-ui/src/main/java/org/xwiki/test/ui/WCAGContext.java

```diff
@@ -0,0 +1,295 @@
+    entry("image-alt", false),
+    entry("input-button-name", false),
+    entry("input-image-alt", false),
+    entry("label", false),
+    entry("link-in-text-block", false),
+    entry("link-name", false),
+    entry("list", false),
+    entry("listitem", false),
+    entry("marquee", false),
+    entry("meta-refresh", false),
+    entry("meta-viewport", false),
+    entry("nested-interactive", false),
+    entry("no-autoplay-audio", false),
+    entry("object-alt", false),
+    entry("role-img-alt", false),
+    entry("scrollable-region-focusable", false),
+    entry("select-name", false),
+    entry("server-side-image-map", false),
+    entry("svg-img-alt", false),
+    entry("td-headers-attr", false),
+    entry("th-has-data-cells", false),
+    entry("valid-lang", false),
+    entry("video-caption", false),
+
+    // WCAG 2.1 Level A & AA Rules
+    entry("autocomplete-valid", false),
+    entry("avoid-inline-spacing", false),
+
+    // WCAG 2.2 Level A & AA Rules
+    entry("target-size", false)
+  );
+  /**
+   *
+   */
+  private static final List<String> VALIDATE_TAGS = Arrays.asList("wcag2a", "wcag2aa", "wcag21a", "wcag21aa");
+  private static final List<String> DISABLED_RULES = Arrays.asList();
+
+  private static class WCAGTestResults {
+    private String failReport = "";
+    private String warnReport = "";
+    private WCAGTestResults(XWikiWebDriver driver, Class validationClass, Results axeResults) {
+      // Generate the report as soon as the results are in.
+      List<Rule> failingViolations = axeResults.getViolations()
+        .stream()
+        .filter(rule -> !FAILS_ON_RULE.containsKey(rule.getId()) || FAILS_ON_RULE.get(rule.getId()))
+        /* If the ruleid is not defined in FAILS_ON_RULE, the default behavior will be to add it to the fails.
+        In order to resolve these test-suite fails quickly, set them as "false" in FAILS_ON_RULE.
+         */
+        .collect(Collectors.toList());
+      if(failingViolations!=null && failingViolations.size()!=0) {
+        AxeReporter.getReadableAxeResults(String.format("WCAG fails on %s", validationClass.getName()),
+          driver, failingViolations);
+        this.failReport = AxeReporter.getAxeResultString();
+      }
+      List<Rule> warningViolations = axeResults.getViolations()
+        .stream()
+        .filter(rule -> FAILS_ON_RULE.containsKey(rule.getId()) && !FAILS_ON_RULE.get(rule.getId()))
+        .collect(Collectors.toList());
+      if (warningViolations!=null && warningViolations.size()!=0) {
+        AxeReporter.getReadableAxeResults(String.format("WCAG warnings on %s", validationClass.getName()),
+          driver, warningViolations);
+        this.warnReport = AxeReporter.getAxeResultString();
+      }
+    }
+    public String getFailReport() {
+      return this.failReport;
+    }
+    public String getWarnReport() {
+      return this.warnReport;
+    }
+    protected boolean failIsEmpty() {
+      return (this.failReport.equals(""));
```

Comment:
The `()` around the return value aren't necessary and should be omitted as there is no good reason to have them.